compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/LangLibrary.java

```diff
@@ -0,0 +1,222 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.compiler.api.impl;
+
+import io.ballerina.compiler.api.impl.types.TypeBuilder;
+import io.ballerina.compiler.api.impl.types.util.BallerinaLangLibMethod;
+import io.ballerina.compiler.api.types.FunctionTypeDescriptor;
+import io.ballerina.compiler.api.types.TypeDescKind;
+import io.ballerina.compiler.api.types.util.LangLibMethod;
+import org.ballerinalang.model.elements.PackageID;
+import org.ballerinalang.model.symbols.SymbolKind;
+import org.wso2.ballerinalang.compiler.semantics.analyzer.Types;
+import org.wso2.ballerinalang.compiler.semantics.model.Scope;
+import org.wso2.ballerinalang.compiler.semantics.model.SymbolEnv;
+import org.wso2.ballerinalang.compiler.semantics.model.SymbolTable;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BInvokableSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BPackageSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.Symbols;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.Name;
+import org.wso2.ballerinalang.compiler.util.Names;
+import org.wso2.ballerinalang.util.Flags;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.compiler.api.types.TypeDescKind.ARRAY;
+import static io.ballerina.compiler.api.types.TypeDescKind.BOOLEAN;
+import static io.ballerina.compiler.api.types.TypeDescKind.DECIMAL;
+import static io.ballerina.compiler.api.types.TypeDescKind.ERROR;
+import static io.ballerina.compiler.api.types.TypeDescKind.FLOAT;
+import static io.ballerina.compiler.api.types.TypeDescKind.FUTURE;
+import static io.ballerina.compiler.api.types.TypeDescKind.INT;
+import static io.ballerina.compiler.api.types.TypeDescKind.MAP;
+import static io.ballerina.compiler.api.types.TypeDescKind.OBJECT;
+import static io.ballerina.compiler.api.types.TypeDescKind.STREAM;
+import static io.ballerina.compiler.api.types.TypeDescKind.STRING;
+import static io.ballerina.compiler.api.types.TypeDescKind.TYPEDESC;
+import static io.ballerina.compiler.api.types.TypeDescKind.XML;
+
+/**
+ * A class to hold the lang library function info required for types.
+ *
+ * @since 2.0.0
+ */
+public class LangLibrary {
+
+    private static final LangLibrary langLibrary = new LangLibrary();
+    private static final String LANG_VALUE = "value";
+    private static Map<String, BPackageSymbol> langLibs;
+    private static Map<String, Map<String, BInvokableSymbol>> langLibMethods;
+    private static Map<String, List<LangLibMethod>> wrappedLangLibMethods;
+    private static TypeBuilder typeBuilder;
```

Comment:
These are static since they are used in a static method

Reply:
I think that is because we are not doing those initializations in the constructor.

We should probably move those to the constructor, and do:
```java
if (langLibrary == null) {
    langLibrary = new LangLibrary();
}

return langLibrary;
```
in the `getInstance()` method